crud/stats/local_registry.lua

```diff
@@ -0,0 +1,124 @@
+local dev_checks = require('crud.common.dev_checks')
+local registry_common = require('crud.stats.registry_common')
+
+local registry = {}
+local internal_registry = {}
+
+--- Initialize local metrics registry
+--
+--  Registries are not meant to used explicitly
+--  by users, init is not guaranteed to be idempotent.
+--
+-- @function init
+--
+-- @treturn boolean Returns true.
+--
+function registry.init()
+    internal_registry.spaces = {}
+    internal_registry.space_not_found = 0
+
+    return true
+end
+
+--- Destroy local metrics registry
+--
+--  Registries are not meant to used explicitly
+--  by users, destroy is not guaranteed to be idempotent.
+--
+-- @function destroy
+--
+-- @treturn boolean Returns true.
+--
+function registry.destroy()
+    internal_registry = {}
+
+    return true
+end
+
+--- Get copy of local metrics registry
+--
+--  Registries are not meant to used explicitly
+--  by users, get is not guaranteed to work without init.
+--
+-- @function get
+--
+-- @tparam string space_name
+--  (Optional) If specified, returns table with statistics
+--  of operations on table, separated by operation type and
+--  execution status. If there wasn't any requests for table,
+--  returns {}. In not specified, returns table with statistics
+--  about all existing spaces and count of calls to spaces
+--  that wasn't found.
+--
+-- @treturn table Returns copy of metrics registry (or registry section).
+--
+function registry.get(space_name)
+    dev_checks('?string')
+
+    if space_name ~= nil then
+        return table.deepcopy(internal_registry.spaces[space_name]) or {}
+    end
+
+    return table.deepcopy(internal_registry)
+end
+
+--- Check if space statistics are present in registry
+--
+-- @function is_unknown_space
+--
+-- @tparam string space_name
+--  Name of space.
+--
+-- @treturn boolean True, if space stats found. False otherwise.
+--
+function registry.is_unknown_space(space_name)
+    dev_checks('string')
+
+    return internal_registry.spaces[space_name] == nil
```

Comment:
Will it work if user creates spaces dynamically? Such caches is a bit dangerous you should carefully invalidate it.

Reply:
I think there is already a careful validation. Maybe code is a bit confusing, so I will describe it with words.

`unknown spaces` case was considered to not create useless stats section if there are some typo in space request or DoS attack. If there is no errors in requests, the new section will be added if it not already exist. Every call to non-existent space will cause an error, so non-existent spaces should not appear here. Call to a newly created dynamic space will create a new section, if it successfull. Then, if there are no stats section and we've got a "space not found" error, new section will not be created and `space_not_found` will increment. If there is not a "space not found" error and space is unknown to statistics registry, statistics module will explicitly call `is_space_exist` util and validate space existence. If we've got an error for existing space, section will be created. So, how it works overall.

- If there are any observations, section already present. It also means that space was existing at least in the past.
- If there are no observations and request is success, or request returned any error and we're sure space exist, we will create a section.
- If there are no observations and request returned "Space not exists" error, or request returned any error we're sure space not exist, we will not create a section.

There is one more corner case. If space was deleted in runtime, errors for this space will be a `space_name.OPERATION.error` and not a `space_not_found` error. Since it is not a typo and not a DoS attack, I think it's ok.

This approach helps to not create excessive load for error content validation or space existence validation.

Reply:
On the other hand, there are no test case for dynamic space create/remove, I should add them.

Reply:
Added test cases for runtime space create/drop.